### ISSUE TITLE
Social Oauth stuck at consent screen

### ISSUE THREAD

**Issue by ssarkar551** [aws-amplify/amplify-flutter#3467]
I am trying to implement a google/facebook auth in flutter. This was working till last week, and now every time an user tries to sign up/ sign in, the screen is stuck in the consent screen, like this: 
![image](https://github.com/aws-amplify/amplify-flutter/assets/46297022/2de22fc4-fb3b-4f69-a057-af2469be6552)

Here's the code that I am using: 
```//google signin flow
  Future<void> googleSignIn({required context}) async {
    try {
      print('in try block google');
      final result = await Amplify.Auth.signInWithWebUI(
        provider: AuthProvider.google,
      );
      print('inside try google sign in');
      Navigator.push(
          context, MaterialPageRoute(builder: (context) => const WelcomePage()));
      safePrint('Sign in result: $result');
    } on AuthException catch (e) {
      safePrint('Error signing in: ${e.message}');
    }
  }```

Note: All client ids and client secrets are still valid, so the problem is not there

**Comment by dnys1** [comment]
Hi @ssarkar551, I believe this is happening to an issue with our implementation of Hosted UI redirect on Android. I've put up a fix in this PR: https://github.com/aws-amplify/amplify-flutter/pull/3496.

If you would like to try it before it's released, you can add a dependency override like so:

```yaml
dependency_overrides:
  amplify_auth_cognito:
    git:
      url: https://github.com/aws-amplify/amplify-flutter.git
      ref: fix/auth/android-hosted-flakiness
      path: packages/auth/amplify_auth_cognito
  amplify_core: ^1.3.0
  amplify_flutter: ^1.3.0
```

Then run `flutter clean && flutter pub upgrade` before running again.

Thanks for your patience resolving this issue.

**Comment by dnys1** [comment]
This has been fixed in v1.3.1. Thanks for your patience resolving this issue.

**Comment by Baaaaatz** [comment]
I am experiencing this issue in my google auth right now. The account is reflecting in user pool but no action is happening after. I believe the scheme I wrote in AndroidManifest.xml is correct because thats the uri in Hosted UI.

There are dependencies I am using.
  amplify_api: 1.3.2
  amplify_auth_cognito: 1.3.3
  amplify_authenticator: 1.3.2
  amplify_flutter: 1.3.3
